src/rollup/plugins/externals.ts

```diff
@@ -239,34 +239,45 @@ export function externals(opts: NodeExternalsOptions): Plugin {
         versions: Record<
           string,
           {
+            pkgJSON: Record<string, any>;
             path: string;
             files: string[];
           }
         >;
       };
       const tracedPackages: Record<string, TracedPackage> = {};
-      await Promise.all(
-        Object.values(tracedFiles).map(async (tracedFile) => {
-          const pkgJSON = await getPackageJson(tracedFile.pkgPath);
-          const pkgName = tracedFile.pkgName; // Use file path as name to support aliases
-          let tracedPackage = tracedPackages[pkgName];
-          if (!tracedPackage) {
-            tracedPackage = {
-              name: pkgName,
-              versions: {},
-            };
-            tracedPackages[pkgName] = tracedPackage;
-          }
-          let tracedPackageVersion = tracedPackage.versions[pkgJSON.version];
-          if (!tracedPackageVersion) {
-            tracedPackageVersion = { path: tracedFile.pkgPath, files: [] };
-            tracedPackage.versions[pkgJSON.version] = tracedPackageVersion;
-          }
-          tracedPackageVersion.files.push(tracedFile.path);
-          tracedFile.pkgName = pkgName;
-          tracedFile.pkgVersion = pkgJSON.version;
-        })
-      );
+      for (const tracedFile of Object.values(tracedFiles)) {
```

Comment:
is it intentional that this is not running in parallel anymore?

Reply:
Yes we need to sync virtual packages and include all traced files.

Reply:
Copy is still parallel